### PR TITLE
feat: Update ES Syntax

### PR DESCRIPTION
**Pull request by scagood**
This extracts the es-syntax data into a separate json file and updates all of it to be up-to-date!

### CODE REVIEW

File: lib/rules/no-unsupported-features/es-syntax-additions.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (scagood) [inline lib/rules/no-unsupported-features/es-syntax-additions.json:1]: This is currently entirely unused. I wanted feedback on the other changes first!

File: lib/rules/no-unsupported-features/es-syntax.json [hunk 2]
```diff
@@ -0,0 +1,607 @@
+{
+    "no-accessor-properties": {
+        "supported": ">=0.10.0"
```
Comment (scagood) [inline lib/rules/no-unsupported-features/es-syntax.json:3]: I changed "supported" to always be a `semver.Range` it makes the comparisons much simpler. I am also thinking about changing the node-builtins to do the same :thinking:

But that is a later monster :joy:

File: tests/lib/rules/no-unsupported-features/es-syntax.js [hunk 3]
```diff
@@ -876,15 +1036,19 @@ runTests([
...
+                            version: "10.0.0",
+                        },
                     },
                 ],
             },
         ],
     },
     {
-        keyword: "new.target",
+        keyword: "newTarget",
```
Comment (scagood) [inline tests/lib/rules/no-unsupported-features/es-syntax.js:1051]: There are a couple of these that I changed, this is in order to match the rule name in `eslint-plugin-es-x`. I was thinking of adding an alias mechanism here, but I think that may get out of hand a little.
| Before | After |
| --- | --- |
| `new.target` | `newTarget` |
| `regexpLookbehind` | `regexpLookbehindAssertions` |
| `regexpS` | `regexpSFlag` |
| `regexpU` | `regexpUFlag` |
| `regexpUnicodeProperties` | `regexpUnicodePropertyEscapes` |
| `regexpY` | `regexpYFlag` |
| `trailingCommasInFunctions` | `trailingFunctionCommas` |
| `unicodeCodePointEscapes` | `unicodeCodepointEscapes` |

File: tests/lib/rules/no-unsupported-features/es-syntax.js [hunk 4]
```diff
@@ -1112,55 +1308,60 @@ runTests([
...
+                        messageId: "not-supported-till",
+                        data: {
+                            ruleId: "no-property-shorthands",
+                            supported: ">=4.0.0",
+                            version: "3.9.9",
+                        },
                     },
                 ],
             },
+            // TODO: Additional es-syntax
```
Comment (scagood) [inline tests/lib/rules/no-unsupported-features/es-syntax.js:1320]: I have commented out `no-property-shorthands-getset`, and `no-bigint-property-names`. as I want to cover them in a less custom way.

I am open to ideas/suggestions

**Comment by scagood** [comment]
There is one big thing that I dont like about this code in its current state, which is the error message:
```
'{{ruleId}}' is not supported until Node.js {{supported}}. The configured version range is '{{version}}'.
```

This comes out to:
```
'no-generators' is not supported until Node.js >=4.0.0. The configured version range is '3.9.9'.
```

My brain can work out what order to place the words, but I hope the code at least makes sense! :joy:

**Comment by aladdin-add** [comment]
shouldn't be `'generators' is not supported until Node.js >=4.0.0. The configured version range is '3.9.9'.`?

```js
ruleId = ruleId.replace(/^no-/, '')
```

File: lib/rules/no-unsupported-features/es-syntax.js [hunk 5]
```diff
@@ -4,428 +4,40 @@
...
-                messageId: "no-numeric-separators",
-            },
-        ],
-    },
-}
-const keywords = Object.keys(features)
+/**
+ * @type {{ruleId: string, supported: import("semver").Range, deprecated: boolean}[]}
+ */
+const features = JSON.parse(readFileSync(resolve(__dirname, "es-syntax.json")))
```
Comment (aladdin-add) [inline lib/rules/no-unsupported-features/es-syntax.js:20]: is there any reason not just using `require(./es-syntax.json)`?

File: lib/rules/no-unsupported-features/es-syntax.js [hunk 6]
```diff
@@ -4,428 +4,40 @@
...
+/** @type {Set<string>} */
+const ignoreKeys = new Set()
+const ruleMap = Object.entries(features).map(([ruleId, meta]) => {
+    const ruleIdNegated = ruleId.slice(3)
```
Comment (aladdin-add) [inline lib/rules/no-unsupported-features/es-syntax.js:24]: `.slice(3)` was to remove the `"no-"`?  good to have a comment for explanation here. :)

File: tests/lib/rules/no-unsupported-features/es-syntax.js (hunk 3 above)
Comment (aladdin-add) [reply tests/lib/rules/no-unsupported-features/es-syntax.js:1050]: I like to be having the same name with the plugin, but, for compat, we may need to a layer to normalize the current name to the new one.

File: lib/rules/no-unsupported-features/es-syntax.js [hunk 7]
```diff
@@ -559,131 +149,22 @@ module.exports = {
...
-
-            //------------------------------------------------------------------
-            // ES2021
-            //------------------------------------------------------------------
-            "no-logical-assignment-operators":
-                "Logical assignment operators are not supported until Node.js {{supported}}. The configured version range is '{{version}}'.",
-            "no-numeric-separators":
-                "Numeric separators are not supported until Node.js {{supported}}. The configured version range is '{{version}}'.",
+            "not-supported-till": [
+                "'{{ruleId}}' is not supported until Node.js {{supported}}.",
```
Comment (aladdin-add) [inline lib/rules/no-unsupported-features/es-syntax.js:161]: nits: the `{{ruleId}}` is a bit confusing, maybe `{{feature}}` or something is better?

File: lib/rules/no-unsupported-features/es-syntax.js (hunk 5 above)
Comment (scagood) [reply lib/rules/no-unsupported-features/es-syntax.js:20]: Nope, my brain was in esm mode! :eyes:

File: lib/rules/no-unsupported-features/es-syntax.js (hunk 7 above)
Comment (scagood) [reply lib/rules/no-unsupported-features/es-syntax.js:161]: Yeah, that would make sense!

**Review by aladdin-add — APPROVED** [review]
👍 LGTM, thanks!

**Pull request merged**